src/validate_constraints.c

```diff
@@ -1898,25 +2001,49 @@ cnode_canonify_propset(struct jvst_cnode *top)
 		struct jvst_cnode_matchset *mset;
 		struct fsm *pat;
 		struct json_str_iter siter;
+		enum re_dialect dialect;
 		struct re_err err = { 0 };
 
 		cons = pm->u.prop_match.constraint;
 		mset = cnode_matchset_new(pm->u.prop_match.match, NULL);
 		mcase = cnode_new_mcase(mset, cons);
+		assert(mcase->next == NULL);
 
 		siter.str = pm->u.prop_match.match.str;
 		siter.off = 0;
+		dialect = pm->u.prop_match.match.dialect;
 
 		pat = re_comp(
-			pm->u.prop_match.match.dialect,
+			dialect,
 			json_str_getc,
 			&siter,
 			opts,
 			(enum re_flags)0,
 			&err);
 
-		// errors should be checked in parsing
-		assert(pat != NULL);
+		if (pat == NULL) {
+			struct json_string str;
+			char tmp[128], *pbuf;
+
+			str = pm->u.prop_match.match.str;
+			if (str.len < sizeof tmp) {
+				pbuf = &tmp[0];
+			} else {
+				pbuf = xmalloc(str.len+1);
+			}
+
+			memcpy(pbuf, str.s, str.len);
+			pbuf[str.len] = '\0';
+
+			re_perror(dialect, &err, NULL, pbuf);
+			if (pbuf != &tmp[0]) {
+				// it's the small diligences, like
+				// free()'ing a temporary buffer, even
+				// though you're going to abort()
```

Comment:
Tempting fate! If someone's reading this comment, they're looking at you're code because they're debugging something.

Reply:
Yeah, I should edit it out.  Not sure why I was being flippant here.